kotlinx-coroutines-core/native/src/MultithreadedDispatchers.kt

```diff
@@ -73,38 +74,80 @@ private class MultiWorkerDispatcher(
     workersCount: Int
 ) : CloseableCoroutineDispatcher() {
     private val tasksQueue = Channel<Runnable>(Channel.UNLIMITED)
+    private val availableWorkers = Channel<CancellableContinuation<Runnable>>(Channel.UNLIMITED)
```

Comment:
Shouldn't its capacity be limited with `workersCount`?

Reply:
It can. Would it improve something? I wanted the code to say "we don't want to ever drop anything here", and "buffer size N" does imply "drop if larger than N."

Reply:
Nothing really, was rather surprised that it has unlimited, while effectively never having more than `workersCount` elements/waiters.

>we don't want to ever drop anything here

Maybe its worth adding asserts/postconditions to each `trySend`?

Reply:
I added some checks. Are they what you meant?

Reply:
That's it, thanks